examples/step-45/doc/intro.dox

```diff
@@ -1,72 +1,142 @@
-respect to the periodicity. We will then write that $j=\text{periodic}(i)$
-(and, if you want, $i=\text{periodic}(j)$).
-If now $U_k, k=0,\ldots,N-1,$ are the unknowns of our discretized problem, then
-the periodic boundary condition boils down to the following set of
-constraints:
+If we instead consider the parallelogram given by the convex hull of
+$(0,0)$, $(1,1)$, $(1,2)$, $(0,1)$ we can achieve the constraints
+$u(0,y)=u(1,y+1)$ by specifying an @p offset:
+@code
+GridTools::collect_periodic_faces(dof_handler,
+                                  /*b_id1*/ 0,
+                                  /*b_id2*/ 1,
+                                  /*direction*/ 0,
+                                  matched_pairs,
+                                  Tensor<1, 2>(0.,1.));
+@endcode
+or
+@code
+GridTools::collect_periodic_faces(dof_handler,
+                                  /*b_id1*/ 0,
+                                  /*b_id2*/ 1,
+                                  /*arbitrary direction*/ 0,
+                                  matched_pairs,
+                                  Tensor<1, 2>(1.,1.));
+@endcode
+
+The resulting matched_pairs can be used in
+DoFTools::make_periodicity_constraints for populating a ConstraintMatrix
+with periodicity constraints:
+@code
+DoFTools::make_periodicity_constraints(matched_pairs, constraints);
+@endcode
+
+Apart from this high level interface there are also variants of
+DoFTools::make_periodicity_constraints available that combine those two
+steps (see the variants of DofTools::make_periodicity_constraints).
+
+There is also a low level interface to
+DofTools::make_periodicity_constraints if more flexibility is needed. The
+low level variant allows to directly specify two faces that shall be
+constrained:
+@code
+using namespace DoFTools;
+make_periodicity_constraints(face_1,
+                             face_2,
+                             constraint_matrix,
+                             component_mask = <default value>;
+                             face_orientation = <default value>,
+                             face_flip = <default value>,
+                             face_rotation = <default value>,
+                             matrix = <default value>);
+@endcode
+Here, we need to specify the orientation of the two faces using
+face_orientation, face_flip and face_orientation. For a closer desription
+have look at the documentation of DoFTools::make_periodicity_constraints.
+The remaining parameters are the same as for the high level interface apart
+from the self-explaining component_mask and constraint_matrix.
+
+<a name="problem"></a>
+<h1>A practical example</h1>
+
+In the following, we show how to use the above functions in a more involved
+example. The task is to enforce rotated periodicity constraints for the
+velocity component of a Stokes flow.
+
+On the domain $\Omega=\{{\bf x}\in[0,1]^2|\|{\bf x}\|\in [0.5,1]\}$ we are
```

Comment:
I think the more widely used notation (at least internationally) is to say "{ x in O : ||x|| ...}" instead of "{ x in O | ||x|| ...}". It is certainly easier to read than the three vertical bars next to each other.

I have a hard time visualizing this domain. Could you add a sketch of it here?